tempodb/backend/azure/azure.go

```diff
@@ -198,6 +211,44 @@ func (rw *readerWriter) ReadRange(ctx context.Context, name string, keypath back
 func (rw *readerWriter) Shutdown() {
 }
 
+func (rw *readerWriter) WriteVersioned(ctx context.Context, name string, keypath backend.KeyPath, data io.Reader, version backend.Version) (backend.Version, error) {
+	// TODO use conditional if-match API
```

Comment:
To properly implement this we need to upgrade our Azure SDK from [azure-storage-blob-go](https://github.com/Azure/azure-storage-blob-go) to the newer [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go). Since this impacts the entire package I'm going to do this in a separate PR...

Reply:
https://github.com/grafana/tempo/issues/2705